File: server/src/main/webapp/WEB-INF/rails/app/presenters/api_v1/notification_filters_representer.rb
#
# Copyright 2019 ThoughtWorks, Inc.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#

module ApiV1
  class NotificationFiltersRepresenter < ApiV1::BaseRepresenter

    FilterStruct = Struct.new(:id, :pipelineName, :stageName, :event, :myCheckin)

    def initialize(filters)
      @represented = Struct.new(:filters).new filters.to_a.map { |nf| FilterStruct.new(*Hash(nf.toMap).symbolize_keys.values_at(*FilterStruct.members)) }
    end

    collection :filters do
      property :id
      property :pipelineName, as: :pipeline
      property :stageName, as: :stage
      property :event
      property :myCheckin, as: :match_commits
    end

  end
end

File: api/api-notification-filter-v2/src/main/java/com/thoughtworks/go/apiv2/notificationfilter/representers/NotificationFiltersRepresenter.java
/*
 * Copyright 2019 ThoughtWorks, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.thoughtworks.go.apiv2.notificationfilter.representers;

import com.thoughtworks.go.api.base.OutputWriter;
import com.thoughtworks.go.domain.NotificationFilter;
import com.thoughtworks.go.spark.Routes;

import java.util.List;

import static com.thoughtworks.go.CurrentGoCDVersion.apiDocsUrl;

public class NotificationFiltersRepresenter {
    public static void toJSON(OutputWriter writer, List<NotificationFilter> notificationFilters) {
        writer.addLinks(outputLinkWriter -> {
            outputLinkWriter.addLink("self", Routes.NotificationFilterAPI.API_BASE);
            outputLinkWriter.addAbsoluteLink("doc", apiDocsUrl("#notification-filters"));
        });

        writer.addEmbedded(embeddedWriter -> {
            embeddedWriter.addChildList("filters", listWriter -> notificationFilters
                    .forEach(notificationFilter -> listWriter.addChild(childWriter -> NotificationFilterRepresenter.toJSON(childWriter, notificationFilter))));
        });
    }
}
